src/openforms/js/components/admin/forms/MappingArrayInput.js

```diff
@@ -0,0 +1,165 @@
+import PropTypes from 'prop-types';
+import React, {useState} from 'react';
+import {FormattedMessage, useIntl} from 'react-intl';
+
+import DeleteIcon from 'components/admin/DeleteIcon';
+import {useOnChanged} from 'components/admin/form_design/logic/hooks';
+
+import ArrayInput from './ArrayInput';
+
+const MappingArrayInput = ({
+  name,
+  inputType,
+  mapping = null,
+  onChange,
+  deleteConfirmationMessage,
+  addButtonMessage,
+  valueArrayInput = false,
+  ...extraProps
+}) => {
+  const intl = useIntl();
+
+  if (!mapping) {
+    mapping = valueArrayInput ? [['', ['']]] : [['', '']];
+  }
+
+  let inputs = [...mapping];
```

Comment:
Initially I had something like this `const [inputs, setInputs] = useState([...mapping])` here, but I ran into the issue that I wanted to be able to manage the state from the `ServiceFetchConfigurationPicker` when I select an existing service fetch configuration